plugin/Utility/SymmetricEncryption.php

```diff
@@ -6,44 +6,55 @@
 
 class SymmetricEncryption
 {
-	const CIPHER = 'AES-128-CBC';
-	private static $key = ONOFFICE_CREDENTIALS_ENC_KEY;
 	const IV_ERROR_MESSAGE = 'iv is false';
 	const DECRYPTION_ERROR_MESSAGE = 'Decryption error';
 
-	public static function encrypt($plainText): string
+	/**
+	 * @param string $plainText
+	 * @param string $key
+	 * @param string $cipher
+	 * @return string
+	 */
+	public function encrypt(string $plainText, string $key, string $cipher = 'AES-128-CBC'): string
 	{
 		if (!extension_loaded('openssl')) {
```

Comment:
Now that the `SymmetricEncryption` class is more generic, just returning the `$plainText` is dangerous. An Exception indicating the OpenSSL extension is not available could be thrown and the fallback case could then be handled externally.

Reply:
ok I will fix that. Thank for your response.

Reply:
I updated as you mentioned: 
 - If the ONOFFICE_CREDENTIALS_ENC_KEY does not exist, It will skipped the encryption.
 - If the OpenSSL is not installed. It will throw an exception.